contrib/envoyproxy/envoy/fakehttp.go

```diff
@@ -0,0 +1,189 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024 Datadog, Inc.
+
+package envoy
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+
+	corev3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	extproc "github.com/envoyproxy/go-control-plane/envoy/service/ext_proc/v3"
+	"google.golang.org/grpc/metadata"
+)
+
+// checkPseudoRequestHeaders Verify the required HTTP2 headers are present
+// Some mandatory headers need to be set. It can happen when it wasn't a real HTTP2 request sent by Envoy,
+func checkPseudoRequestHeaders(headers map[string]string) error {
+	for _, header := range []string{":authority", ":scheme", ":path", ":method"} {
+		if _, ok := headers[header]; !ok {
+			return fmt.Errorf("missing required headers: %q", header)
+		}
+	}
+
+	return nil
+}
+
+// checkPseudoResponseHeaders Verify the required HTTP2 headers are present
```

Comment:
```suggestion
// checkPseudoResponseHeaders verifies the required HTTP2 headers are present
```

Reply:
Applied in db73a1f13bf8855a9c1a351a685f38629d5926bc thanks 😄